src/types.rs

```diff
@@ -979,13 +963,20 @@ impl SourceMap {
                     //  Advance the `original_range`.
                     match original_ranges_iter.next() {
                         Some(r) => original_range = r,
-                        None => return builder.into_sourcemap(),
+                        None => break 'outer,
                     }
                 }
             }
         }
 
-        builder.into_sourcemap()
+        // Regenerate the index
+        self.index = self
+            .tokens
+            .iter()
+            .enumerate()
+            .map(|(idx, token)| (token.dst_line, token.dst_col, idx as u32))
+            .collect();
```

Comment:
You did a `self.index.clear` above. If you use `extend` instead of `collect`, you can potentially reuse that allocation.

Reply:
Nice catch!